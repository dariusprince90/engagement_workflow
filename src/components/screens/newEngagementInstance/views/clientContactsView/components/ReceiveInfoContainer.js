import { memo } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const propTypes = {
  receiveCustomerInvoice: PropTypes.bool.isRequired,
  receiveCustomerStatement: PropTypes.bool.isRequired
};

let ReceiveInfoContainer = ({ receiveCustomerInvoice, receiveCustomerStatement }) => {
  // **********************************************************************
  // * constants

  const TRUE_ICON = 'fa-solid fa-check';
  const TRUE_ICON_CLASS = 'text-success';

  const FALSE_ICON = 'fa-solid fa-xmark';
  const FALSE_ICON_CLASS = 'text-danger';

  // **********************************************************************
  // * component vars

  const invoiceIconProps = {
    icon: receiveCustomerInvoice ? TRUE_ICON : FALSE_ICON,
    className: receiveCustomerInvoice ? TRUE_ICON_CLASS : FALSE_ICON_CLASS,
    size: 'lg',
    fixedWidth: true
  };

  const statementIconProps = {
    icon: receiveCustomerStatement ? TRUE_ICON : FALSE_ICON,
    className: receiveCustomerStatement ? TRUE_ICON_CLASS : FALSE_ICON_CLASS,
    size: 'lg',
    fixedWidth: true
  };

  // **********************************************************************
  // * functions

  // **********************************************************************
  // * side effects

  // **********************************************************************
  // * render

  return (
    <div className="receive-info-container">
      <div className="receive-info-item">
        <div>Receive Customer Invoice</div>
        <FontAwesomeIcon {...invoiceIconProps} />
      </div>
      <div className="receive-info-item">
        <div>Receive Customer Statement</div>
        <FontAwesomeIcon {...statementIconProps} />
      </div>
    </div>
  );
};

ReceiveInfoContainer = memo(ReceiveInfoContainer);
ReceiveInfoContainer.propTypes = propTypes;
ReceiveInfoContainer.displayName = 'ReceiveInfoContainer';

export default ReceiveInfoContainer;
