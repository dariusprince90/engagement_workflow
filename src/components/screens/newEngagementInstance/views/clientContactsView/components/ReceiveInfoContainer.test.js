/* eslint-disable react/prop-types */

import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import faker from '@faker-js/faker';

import ReceiveInfoContainer from './ReceiveInfoContainer';

// **********************************************************************
// * constants

const testIds = {
  fontAwesomeIcon: 'font-awesome-icon'
};

const defaultProps = {
  receiveCustomerInvoice: faker.datatype.boolean(),
  receiveCustomerStatement: faker.datatype.boolean()
};

// **********************************************************************
// * functions

const getComponentToRender = (props) => {
  return <ReceiveInfoContainer {...props} />;
};

// **********************************************************************
// * mock external dependencies

jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: ({ icon, className, size, fixedWidth }) => {
    const props = { icon, className, size, fixedWidth };
    return <fake-font-awesome-icon {...props} data-testid={testIds.fontAwesomeIcon} />;
  }
}));

// **********************************************************************
// * unit tests

describe('ReceiveInfoContainer', () => {
  // **********************************************************************
  // * setup

  // **********************************************************************
  // * tear-down

  // **********************************************************************
  // * execution

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponentToRender(defaultProps), div);
  });

  it.each([
    {
      receiveCustomerInvoice: true,
      icon: 'fa-solid fa-check',
      iconClass: 'text-success',
      receiveCustomerStatement: false
    },
    {
      receiveCustomerInvoice: false,
      icon: 'fa-solid fa-xmark',
      iconClass: 'text-danger',
      receiveCustomerStatement: false
    }
  ])(
    'renders the "Receive Customer Invoice" icon with the "$icon" icon and "$iconClass" className when receiveCustomerInvoice is $receiveCustomerInvoice',
    ({ receiveCustomerInvoice, receiveCustomerStatement }) => {
      const props = { ...defaultProps, receiveCustomerInvoice, receiveCustomerStatement };
      const tree = renderer.create(getComponentToRender(props)).toJSON();
      expect(tree).toMatchSnapshot();
    }
  );

  it.each([
    {
      receiveCustomerStatement: true,
      icon: 'fa-solid fa-check',
      iconClass: 'text-success',
      receiveCustomerInvoice: false
    },
    {
      receiveCustomerStatement: false,
      icon: 'fa-solid fa-xmark',
      iconClass: 'text-danger',
      receiveCustomerInvoice: false
    }
  ])(
    'renders the "Receive Customer Statement" icon with the "$icon" icon and "$iconClass" className when receiveCustomerStatement is $receiveCustomerStatement',
    ({ receiveCustomerStatement, receiveCustomerInvoice }) => {
      const props = { ...defaultProps, receiveCustomerStatement, receiveCustomerInvoice };
      const tree = renderer.create(getComponentToRender(props)).toJSON();
      expect(tree).toMatchSnapshot();
    }
  );
});
