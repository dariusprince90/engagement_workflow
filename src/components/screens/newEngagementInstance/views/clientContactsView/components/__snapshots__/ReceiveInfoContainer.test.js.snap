// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiveInfoContainer renders the "Receive Customer Invoice" icon with the "fa-solid fa-check" icon and "text-success" className when receiveCustomerInvoice is true 1`] = `
<div
  className="receive-info-container"
>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Invoice
    </div>
    <fake-font-awesome-icon
      className="text-success"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-check"
      size="lg"
    />
  </div>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Statement
    </div>
    <fake-font-awesome-icon
      className="text-danger"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-xmark"
      size="lg"
    />
  </div>
</div>
`;

exports[`ReceiveInfoContainer renders the "Receive Customer Invoice" icon with the "fa-solid fa-xmark" icon and "text-danger" className when receiveCustomerInvoice is false 1`] = `
<div
  className="receive-info-container"
>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Invoice
    </div>
    <fake-font-awesome-icon
      className="text-danger"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-xmark"
      size="lg"
    />
  </div>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Statement
    </div>
    <fake-font-awesome-icon
      className="text-danger"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-xmark"
      size="lg"
    />
  </div>
</div>
`;

exports[`ReceiveInfoContainer renders the "Receive Customer Statement" icon with the "fa-solid fa-check" icon and "text-success" className when receiveCustomerStatement is true 1`] = `
<div
  className="receive-info-container"
>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Invoice
    </div>
    <fake-font-awesome-icon
      className="text-danger"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-xmark"
      size="lg"
    />
  </div>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Statement
    </div>
    <fake-font-awesome-icon
      className="text-success"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-check"
      size="lg"
    />
  </div>
</div>
`;

exports[`ReceiveInfoContainer renders the "Receive Customer Statement" icon with the "fa-solid fa-xmark" icon and "text-danger" className when receiveCustomerStatement is false 1`] = `
<div
  className="receive-info-container"
>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Invoice
    </div>
    <fake-font-awesome-icon
      className="text-danger"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-xmark"
      size="lg"
    />
  </div>
  <div
    className="receive-info-item"
  >
    <div>
      Receive Customer Statement
    </div>
    <fake-font-awesome-icon
      className="text-danger"
      data-testid="font-awesome-icon"
      fixedWidth={true}
      icon="fa-solid fa-xmark"
      size="lg"
    />
  </div>
</div>
`;
